drivers/flash/spi_nor.c

```diff
@@ -909,6 +915,12 @@ static int spi_nor_configure(const struct device *dev)
 		return -ENODEV;
 	}
 
+#if defined(CONFIG_SPI_NOR_UNLOCK_ON_STARTUP)
```

Comment:
If some device needs this for basic functioning it should be a devicetree property, not a Kconfig.  It would presumably also apply to the other SPI NOR drivers so belongs in `dts/bindings/mtd/jedec,spi-nor-common.yaml`.

What device motivates this change, and how?  Is it one that has volatile BP bits that power-up in write-protected mode?  Because some devices have these bits non-volatile, and making it too easy to blindly remove write protection could cause problems.

At a minimum we should be looking to JESD216 DW1's "Volatile Status Register Block Protect bits" flag, and to DW16's "Volatile or Non-Volatile Register and Write Enable Instruction for Status Register 1" to determine when this is necessary.  Also care should be taken to preserve the values of any (possibly non-volatile) SR1 bits that are not related to block protection.

Reply:
@pabigot yes I have a device that powers up as write protected. I just copied the way that this is handled in linux kernel: https://github.com/torvalds/linux/blob/master/drivers/mtd/spi-nor/core.c#L3206

Reply:
What device do you have?

Reply:
SST25VF020B

Reply:
Please build and run `samples/drivers/jesd216` on your board and post the output here.

Reply:
@pabigot I get the same as @ycsin. The reason is that the device has no read sfdp command

Reply:
OK, I see that.  Looks like this sort of behavior was mostly pre-JESD216.

Please try the solution in #32680.  There you specify in devicetree that the device needs certain bits cleared during initialization, and only those bits are cleared.